software/controller/lib/core/v03system.cpp

```diff
@@ -0,0 +1,180 @@
+                       GPIO::AnalogInputPin(adc_channel(Sensor::FIO2), &adc));
+  air_influx_sensor_dp_.emplace(
+      "air_influx_", "for ambient air influx",
+      GPIO::AnalogInputPin(adc_channel(Sensor::AirInflowPressureDiff), &adc), ADC::VoltageRange);
+  oxygen_influx_sensor_dp_.emplace(
+      "oxygen_influx_", "for concentrated oxygen influx",
+      GPIO::AnalogInputPin(adc_channel(Sensor::OxygenInflowPressureDiff), &adc), ADC::VoltageRange);
+  outflow_sensor_dp_.emplace("outflow_", "for outflow",
+                             GPIO::AnalogInputPin(adc_channel(Sensor::OutflowPressureDiff), &adc),
+                             ADC::VoltageRange);
+
+  // These require existing DP sensors to link to
+  air_influx_sensor_.emplace("air_influx_", "for ambient air influx",
+                             &air_influx_sensor_dp_.value(), VenturiPortDiameter,
+                             VenturiChokeDiameter, VenturiCorrection);
+  oxygen_influx_sensor_.emplace("oxygen_influx_", "for concentrated oxygen influx",
+                                &oxygen_influx_sensor_dp_.value(), VenturiPortDiameter,
+                                VenturiChokeDiameter, VenturiCorrection);
+  outflow_sensor_.emplace("outflow_", "for outflow", &outflow_sensor_dp_.value(),
+                          VenturiPortDiameter, VenturiChokeDiameter, VenturiCorrection);
+}
+
+void V03Sensors::calibrate() {
+  /// \TODO It seems that we'll need to save calibration readings to non-volatile memory and provide
+  //    operators with a way to shut off the device's blowers, open any necessary valves, and
+  //    recalibrate.
+  patient_pressure_sensor_->set_zero();
+  air_influx_sensor_dp_->set_zero();
+  oxygen_influx_sensor_dp_->set_zero();
+  outflow_sensor_dp_->set_zero();
+  fio2_sensor_->set_zero();
+}
+
+/// \TODO Add alarms if sensor value is out of expected range?
+
+SensorReadings V03Sensors::get_readings() const {
+  /// \TODO Potential Caution: Density of air slightly varies over temperature and altitude - need
+  ///   mechanism to adjust based on delivery? Constant involving density of air. Density assumed at
+  ///   15 deg. Celsius and 1 atm of pressure.
+  ///   Sourced from https://en.wikipedia.org/wiki/Density_of_air
+  static constexpr float AirDensity{1.225f};  // kg/m^3
+
+  // Assuming ambient pressure of 101.3 kPa
+  /// \TODO measure ambient pressure from an additional sensor and/or estimate from user input
+  ///   (from altitude?)
+  static constexpr Pressure AmbientPressure = kPa(101.3f);
+
+  return {
+      .patient_pressure = patient_pressure_sensor_->read(),
+      .fio2 = fio2_sensor_->read(AmbientPressure),
+      .air_inflow = air_influx_sensor_->read(AirDensity),
+      .oxygen_inflow = oxygen_influx_sensor_->read(AirDensity),
+      .outflow = outflow_sensor_->read(AirDensity),
+  };
+}
+
+V03Actuators::V03Actuators()
+    : blower_pinch_("blower_valve_", " of the blower pinch valve", BlowerValveMotorIndex),
```

Comment:
Actually, I made both pinch valves `std::optional` now and they initialize in `init()`.
We want to keep all this in the cpp, because I'd rather include `system_constants.h` here and leave the definition as generic as possible.